internal/pkg/cli/svc_deploy.go

```diff
@@ -328,7 +333,48 @@ func (o *deploySvcOpts) getDockerfilePath() (string, error) {
 	if !ok {
 		return "", fmt.Errorf("service %s does not have a dockerfile path", o.Name)
 	}
-	return mf.DockerfilePath(), nil
+
+	copilotDir, err := o.ws.CopilotDirPath()
+	if err != nil {
+		return "", fmt.Errorf("get workspace root directory: %w", err)
+	}
+	wsRoot := filepath.Dir(copilotDir)
+	return filepath.Join(wsRoot, mf.DockerfilePath()), nil
+}
+
+func (o *deploySvcOpts) getDockerfileContext() (string, error) {
```

Comment:
Hmm can we possibly combine `getDockerfileContext()` and `getDockerfilePath()` to sth like `getDockerfile() (struct, error)`?